feat: new website v2

alexiscolin opened the pull request:
# Gno Website Design V2

This PR aims to deliver a new website layout design by sync the `www.gno.land` repo (used for static contents on Gno main website). It adds more flexibility and (HTML/CSS) components that will be decoupled and reuse in #781  with dedicated markdown markup. 
As of today, changes mainly involve Markdown and CSS files, but some Go templates have been updated (such as header menu, blogs listing...)

- [X] New Fonts
- [X] Navigation
- [X] New Home page
- [X] New Blog page
- [X] New Event page
- [X] New About page
- [X] Packages pages
- [X] Responsive navigation
- [X] Columns (2 / 3) component
- [X] Jumbotron component
- [X] Accordion component
- [X] Tabs component
- [X] Stack component
- [X] New  dedicated GoR page
- [X] Email form design 
- [X] CSS cleanup (variables / relative units ...)

**⚠️  Note:** This is a draft and the render has not been tested everywhere or in every situation yet.

ajnavarro commented:
Would be great to be able to see a render somewhere.

tbruyelle commented on gno.land/cmd/gnoweb/pages/EVENTS.md:
```diff
@@ -0,0 +1,92 @@
+[Learn more](/)
+
+</div>
+</div
```
```suggestion
</div>
```

tbruyelle commented on gno.land/cmd/gnoweb/pages/HOME.md:
```diff
@@ -1,50 +1,81 @@
+
+</div>
+
+<div class="jumbotron">
```
Why not using `::: jumobotron` here ?

tbruyelle replied:
Oh OK sorry I misunderstood, this will be supported in #781

tbruyelle commented:
> Would be great to be able to see a render somewhere.

You can run the website locally:

```bash
$ cd gno.land
$ go run ./cmd/gnoweb
Running on http://127.0.0.1:8888
```

thehowl commented:
>You can run the website locally:

Yes indeed, but for convenience I also made a small screencast of the current state of the website :)

https://drop.howl.moe/e75fb57f97-2023-05-12_11-36-10.mp4

Great work @alexiscolin !

alexiscolin commented:
Here is a new preview (Desktop ver) 

https://github.com/gnolang/gno/assets/7622257/c62e6766-2f78-4051-acec-3146759e4e83

alexiscolin commented:
And the Mobile ver

https://github.com/gnolang/gno/assets/7622257/16800791-121f-46db-8001-e6f3d629e8b5

tbruyelle commented on examples/gno.land/p/demo/blog/blog.gno:
```diff
@@ -209,12 +212,14 @@ func (p *Post) RenderListItem() string {
-	output += ufmt.Sprintf("## [&#9656; %s](%s)\n", p.Title, p.URL())
+	output := "<div>\n\n"
+	output += ufmt.Sprintf("## [%s](%s)\n", p.Title, p.URL())
+	output += ufmt.Sprintf("**[%s](%s)**\n", "Learn More", p.URL())
```
```suggestion
	output += ufmt.Sprintf("**[Learn More](%s)**\n", p.URL())
```

:angel:

tbruyelle commented on gno.land/cmd/gnoweb/pages/HOME.md:
```diff
@@ -1,50 +1,77 @@
-- [About Gno.land](/about)
+---
+
+## An interpretation of the Golang (Go) programming language for advanced developers and intrepid pioneers to build succinct, composable smart contracts for social coordination.
```
Is this from you ? I like to be an intrepid pioneer xD

alexiscolin replied:
If only I could be the writer haha! It's from Christina 😆

tbruyelle commented on gno.land/cmd/gnoweb/pages/HOME.md:
```diff
@@ -1,50 +1,77 @@
+
+## An interpretation of the Golang (Go) programming language for advanced developers and intrepid pioneers to build succinct, composable smart contracts for social coordination.
+
+If you’re concerned about information censorship and want to contribute to the #GnoWorldOrder, follow our socials to find out how
```
```suggestion
If you’re concerned about information censorship and want to contribute to the #GnoWorldOrder, follow our socials to find out how.
```

tbruyelle commented on gno.land/cmd/gnoweb/pages/HOME.md:
```diff
@@ -1,50 +1,77 @@
+### Learn about Gno.land
+
+- [About](/about)
+- [Subscribe](/)
```
Shouldn't you add an anchor to the bottom of the page, where there is the subscribe form ?

alexiscolin replied:
You are right!

tbruyelle commented on gno.land/cmd/gnoweb/pages/HOME.md:
```diff
@@ -1,50 +1,77 @@
+### Explore the universe
+
+- [Discover demo packages](https://github.com/gnolang/gno/tree/master/examples)
+- [Install Gno Key instructions](/r/demo/boards:testboard/5)
```
The instructions in the post are not up-to-date , I will create a separate issue to fix them.

tbruyelle replied:
https://github.com/gnolang/gno/issues/865

tbruyelle commented on gno.land/cmd/gnoweb/pages/HOME.md:
```diff
@@ -1,50 +1,77 @@
+
+### Build with Gnolang
+
+- [Gno dev with CLI](/)
```
link ?

tbruyelle commented on gno.land/cmd/gnoweb/pages/HOME.md:
```diff
@@ -1,50 +1,77 @@
+
+### Explore the universe
+
+- [Discover demo packages](https://github.com/gnolang/gno/tree/master/examples)
```
I wonder if it's not preferable to have a specific page to browse the examples, like we have in the current version of the website.

![image](https://github.com/gnolang/gno/assets/92280/b47dc102-251f-41b2-b371-5c88ba65ed13)

WDYT?

alexiscolin replied:
I think another website dedicated to Gno examples (Gno by Example) will be created soon. Maybe these demo should live there (with a link from this website)?

tbruyelle replied:
Sounds good OK.

tbruyelle reviewed (commented):
Awesome work, I love this new website!

moul commented on gno.land/cmd/gnoweb/pages/HOME.md:
```diff
@@ -1,50 +1,77 @@
+
+[Join Game Of Realms](/game-of-realms)
+
+</div>
+</div>
```
is it expected?

moul replied:
Ok, so yeah make sense to add comments when we have multiple levels.

moul commented on examples/gno.land/p/demo/blog/blog.gno:
```diff
@@ -27,6 +27,7 @@ func (b Blog) Render(path string) string {
 		output := breadcrumb([]string{b.Title})
+		output += "<div class='columns-3'>"
 
 		if b.Posts.Size() == 0 {
```
```go
} else {
	output += "<div class='columns-3'>"
	// etc
}
```

thehowl replied:
It's not correct otherwise. (Actually, I would say simply put `output += "<div class='columns-3'>"` after the end of the if block, as that one returns), because otherwise `<div>` is printed even when there are no posts (and no corresponding `</div>`) - like it happens in gnoblog_test.gno.

alexiscolin replied:
Ha yes, I missed the conditional return. Thank you!

moul commented on gno.land/cmd/gnoweb/pages/EVENTS.md:
```diff
@@ -0,0 +1,114 @@
+
+[Watch the talk](https://www.youtube.com/watch?v=IJ0xel8lr4c)
+
+</div>
+<div>
```
even if it's more code and potentially useless, I think we should encourage naming divs or adding comments.

```suggestion
</div>
<div class="column">
```

or

```suggestion
</div>
<div> <!-- column -->
```

alexiscolin replied:
That's a good idea to improve readability! Let's do it.

moul commented on gno.land/cmd/gnoweb/main.go:
```diff
@@ -124,7 +125,19 @@ func handlerGor(app gotuna.App) http.Handler {
+	})
+}
+
+func handlerEvents(app gotuna.App) http.Handler {
+	md := filepath.Join(flags.pagesDir, "EVENTS.md")
+	mainContent := osm.MustReadFile(md)
+
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		app.NewTemplatingEngine().
+			Set("MainContent", string(mainContent)).
+			Set("Title", "Events").
 			Render(w, r, "generic.html", "funcs.html")
 	})
 }
```
OR, we just make it a smart contract so we can keep it updated.

Can be an extension of `r/gnoland/blog` to improve the experience for non-post pages, or maybe better, a simple new realm: `r/gnoland/pages:events`.

moul commented on gno.land/cmd/gnoweb/pages/GOR.md:
```diff
@@ -1,12 +1,184 @@
 # Game of Realms
```
this should be a contract too.

moul replied:
Will be addressed with #874, you can just remove those files, thank you.

moul commented on gno.land/cmd/gnoweb/static/font/font.css:
```diff
@@ -1,41 +1,43 @@
 
+/**** ROBOTO ****/
```
what's the license for this font?

moul replied:
can you add the LICENSE files or a comment?

thehowl commented on gno.land/cmd/gnoweb/static/js/renderer.js:
```diff
@@ -6,3 +6,190 @@
+    this.tabs = Array.from(this.tablistNode.querySelectorAll("[role=tab]"));
+    this.tabpanels = [];
+
+    for (var i = 0; i < this.tabs.length; i += 1) {
```
`this.tabs.forEach`? or `for (let tab of this.tabs)`

alexiscolin replied:
The first one if you like functional programming and the second one for readability, mine for speed.

alexiscolin replied:
Let use the `for of`

thehowl commented on gno.land/cmd/gnoweb/static/js/renderer.js:
```diff
@@ -6,3 +6,190 @@
+ *   Desc: Tablist widget that implements ARIA Authoring Practices
+ */
+
+class Tabs {
```
also, do we need this? I don't see this being used

alexiscolin replied:
It will be used soon in GoR page (the one I removed to let the content come from realms) 👍

thehowl commented on gno.land/cmd/gnoweb/views/faucet.html:
```diff
@@ -7,35 +7,31 @@
+        <script src="https://www.google.com/recaptcha/api.js" async defer></script>
+        {{ end }}
+        <script type="text/javascript">
+          function main() {
+              // XXX
+          };
```
maybe remove it and remove `onload="main()"` from body?

alexiscolin replied:
You are right, I wanted to create another PR to do a good cleaning/webperf improvements, since there is a lot of old remaining/duplicate code. But let's improve what we can now. Thank you!

thehowl reviewed (commented):
Screencast of my clicking around the new website: https://drop.howl.moe/4f1921404c-2023-06-15_19-59-54.mp4

It looks good, I have some general comments aside the code-specific ones:

1. Can we make the sun/moon icon for the theme switcher the same color as the GitHub icon in the header?
2. I see we render a `<hr>` we don't really use at the bottom. Can we remove it?
3. Maybe we should add some spacing on the `<pre>` / code blocks to make them nicer aesthetically?
4. Should we have something, when viewing a realm render, to visually distinguish the realm and path + source/help links at the top? I know they're already full-size compared to the content -- but maybe make them have a slightly darker background, in transition from the header's background?
5. I think the "Socials" section in the homepage footer has a lot of negative space on desktop. Maybe have the subscribe form on the right?

alexiscolin commented:
@moul I've updated the content as we discussed yesterday. I've added two static pages "ecosystem" and "language" that follow the `generic` template. 

I've also removed the `Events` and `GoR` contents. So I will let you push the data onchain, then I will probably need help to retrieve and display them.  For the record, as I've removed these pages, you can find their display architecture in this [commit](https://github.com/gnolang/gno/pull/818/commits/d6d1d8a4f9663d9cf29aa194c0862d1b1199696c).

Here is the contents:

- for [Events](https://docs.google.com/document/d/1lUczzfRTLtpJvPi8tKWrSYduaWhBhFTUqf1TShThMVk/)
- for [GoR](https://docs.google.com/document/d/1rtGeLygo2gV60Dxt8zY27EraYiP4EXXzs6wDu3wNj4A/)

alexiscolin commented:
@moul for the testnet style merge,

- I've quickly reset the main pages to keep them simple (HOME, GOR...) 
- I've removed all static pages that are not necessary cause embedded intro `www.gno.land` repo
- I've sync the static repo style with this one

Be careful about the style render into blog article since it hasn't been fully tested with long and complex article (with all the exotic `md` syntax yet)

Could be great to take a look at `sources` and `help` content as well. I've checked them but maybe some exotic content I'm not aware of are hidden in the contents.
